docs/source/references/apollo-config.md

```diff
@@ -0,0 +1,201 @@
+  },
+};
+```
+
+### `addTypename`
+
+*Optional* –– Apollo will by default add the `__typename` field to all operations automatically.
+
+GraphQL clients like Apollo Client often add the `__typename` field to operations automatically when they're sent over the wire. This can come in really handy for things like caching, but it can be turned off by adding `addTypename: false` to the client config:
+
+```js line=3
+module.exports = {
+  client: {
+    addTypename: false,
+  },
+};
+```
+
+### `clientOnlyDirectives`, `clientSchemaDirectives`
+
+*Optional* –– By default, Apollo projects support the following client-side directives:
+
+- `@client` for local state
+- `@rest` for using apollo-link-rest
+- `@connection` for custom pagination with Apollo Client
+- `@type` for dynamic type names with apollo-link-rest
+
+Client side applications can use custom directives on their queries that aren't meant to be sent to the server. Configuration of client side directives beyond the defaults listed above can be set up like so:
+
+```js line=3-4
+module.exports = {
+  client: {
+    clientOnlyDirectives: ['connection', 'type'],
+    clientSchemaDirectives: ['client', 'rest'],
+  },
+};
+```
+
+`clientOnlyDirectives` are directives that should be stripped out of the operation before being sent to the server. An example of this is the `@connection` directive.
+
+`clientSchemaDirectives` are directives that indicate a portion of the operation that is not meant to be sent to the server. These directives are removed as well as the fields they are placed on. An example of this type of of directive is the `@client` directive.
+
+<h2 id="service-config">Server projects</h2>
+
+Server projects are configured through a top level `service` key in the config.
+
+```js line=2
+module.exports = {
+  service: { ... },
+};
+```
+
+Setting up a `service` in your Apollo config will save you from needing to use flags your Apollo CLI commands like `apollo service:push` and `apollo service:check`. You can set up the schema for your service to load in one of two ways:
+1. Using a remote endpoint
+1. Using a local schema file
+
+<h4 id="service-remote-endpoint">Option 1: Remote endpoint</h4>
+
+Remote endpoints can be used to pull down a schema from a running service. This can be configured like so:
+
+```js line=2-10
+module.exports = {
+  service: {
+    endpoint: {
+      url: 'https://api.github.com/graphql', // defaults to http://localhost:4000
+      headers: { // optional
+        authorization: 'Bearer lkjfalkfjadkfjeopknavadf',
+      },
+      skipSSLValidation: true, // optional, disables SSL validation check
+    },
+  },
+};
+```
+
+<h4 id="service-local-file">Option 2: Local schema</h4>
+
+In some cases, teams may have a local generated file of their schema that they want to link. This can be either a `.graphql` file with the schma in SDL, or a saved introspection result in `.json`. To link a service with a local file, configure the project like this:
```

Comment:
Going to recommend the same language change as before in the other local schema section.